Core/Src/datalogging/odometer_calc.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * odometer_calc.c
+ *
+ *  Created on: May 11, 2020
+ *      Author: renanmoreira
+ */
+
+#include "datalogging/odometer_calc.h"
+
+#include "cmsis_os.h"
+#include "datalogging/datalog_handler.h"
+#include "datalogging/odometer_save.h"
+#include "main.h"
+#include "util/CMSIS_extra/global_variables_handler.h"
+#include "util/flash_sector_h7.h"
+#include "util/global_definitions.h"
+#include "util/global_instances.h"
+#include "util/util.h"
+
+// Function to calculate distance traveled
+static inline uint32_t calculate_distance(uint32_t speed_avg);
+
+// Function to send data via CAN
+void log_distance(ODOMETER_TOTAL_t total_dist, uint32_t partial_dist);
+
+// Thread
+void odometer_calc()
+
+{
+    // Useful variables declaration
+    uint32_t front_speed_avg       = 0;
+    uint32_t partial_dist_traveled = 0;
+
+    // Odometer variable created
+    ODOMETER_TOTAL_t total_dist_traveled;
+
+    // Read distance from flash just once
+    Flash_Read_Data(ODOMETER_DATA_FLASH_ADDR, &(total_dist_traveled), WORDS_READ_ONE);
+
+    // In case the memory position has never been used/saved
+    if ((total_dist_traveled == 0x00000000) || (total_dist_traveled == 0xFFFFFFFF)) {
+        total_dist_traveled = 0;
+    }
+
+    // Store in the global variable
+    set_global_var(ODOMETER_TOTAL, &total_dist_traveled);
+
+    for (;;) {
+#ifdef DEBUG_ECU
+        extern void brkpt();
+        brkpt();
+#endif
+        // Wait event flag to be set so the task runs
+        osEventFlagsWait(ECU_control_event_id, RTD_FLAG, osFlagsNoClear, osWaitForever);
+
+        // Use global defined struct to read wheel speeds
+        WHEEL_SPEEDS_t wheel_speeds = get_global_var_value(WHEEL_SPEEDS);
+
+        // Calculate the front wheel speed average due to reduced change of locking up
```

Comment:
o que seria “reduced change of locking up”?
(btw esse comentário parece desnecessário também, o propósito do código tá claro embaixo)

Reply:
Era pra ser "reduced (chance) of locking up". Era um comentário pra justificar o cálculo da média de velocidade usando as rodas dianteiras, considerando que elas tem menor chance de travar se comparadas com as traseiras.

Entendi as sugestões sobre os comentários e vou alterar. Escrevi eles durante a implementação mais como forma de me encontrar e organizar as ideias